### PR TITLE
DEV | GenerateData 

### LINKED ISSUE #128: BUG - Simplify Mock Data Usage Directions
After project set up and dockerization generate_data command needs to be run in app server terminal in docker. Readme file to be updated.

### PR DESCRIPTION
**Pull request by sozkan4**
Description 
Mock data models added for 1 admin and 2 normal users for 4 different spaces.

### CODE REVIEW

File: core/management/generate_data.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            Space(name="Cat Memes"),
+            Space(name="Tech Geeks"),
+            Space(name="Pizza Lovers"),
+            Space(name="Coffee Addicts"),
+        ]
+        Space.objects.bulk_create(spaces)
+
+        print("Mock data created.")
+
+for space in spaces:
```
Comment (ceydaduzgec) [inline core/management/generate_data.py]: These should be inside of the handle function. Please correct this and push it again.
If you want to learn more about what we are doing here:
- https://simpleisbetterthancomplex.com/tutorial/2018/08/27/how-to-create-custom-django-management-commands.html
- https://docs.djangoproject.com/en/4.2/howto/custom-management-commands/

File: README.md [hunk 2]
```diff
@@ -122,22 +122,18 @@ This projects is for the SWE 574 Software Development Practice course of Boğazi
...
+<!-- DOCUMENTATION -->
+### Create Mock Data
 
-- [ ] Milestone 1
-- [ ] Milestone 2
-- [ ] Milestone 3
-
-See the [open issues](https://github.com/swe574-spring23/SWE574/issues) for a full list of proposed features (and known issues).
+To generate mock data for your local, you can use the following command, after the installation steps:
+  `python manage.py generate_data.py`
```
Comment (ceydaduzgec) [inline README.md]: I realized I wrote the `.py` extra. It should be `python manage.py generate_data`

File: core/management/generate_data.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+from django.core.management import BaseCommand
```
Comment (ceydaduzgec) [inline core/management/generate_data.py]: Also the file structure should be: `core/management/commands/generate_data.py` 
And `core`app should be included in `INSTALLED_APPS`Can you fix this as well?

File: core/management/generate_data.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        Space.objects.bulk_create(spaces)
+
+        print("Mock data created.")
+
+for space in spaces:
+    space.save()
+
+for _ in range(4):  
+    user = User(
+        email=fake.email(),
```
Comment (ceydaduzgec) [inline core/management/generate_data.py]: Fake library is not included in imports. You can import them here or can use UserFactory to generate random users here.

File: core/management/generate_data.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+for post_data in posts_data:
+    post = Post(
+        author=random.choice(User.objects.all()),
+        space=post_data["space"],
+        title=post_data["title"],
+        content=post_data["content"],
+        link=post_data["link"],
+    )
+    post.save()
+()
```
Comment (ceydaduzgec) [inline core/management/generate_data.py]: This () is extra please delete this and also please use the linter or we cannot merge this pull request without successfully passing the linter check.

**Review by ceydaduzgec — CHANGES_REQUESTED** [review]
Great start but we need some little things that should be fixed. Let me know if something is unclear. Also can you fix the title of the pull request?

**Comment by sozkan4** [comment]
Hello Ceyda I fixed the code and test it working fine also fixed read me file. One concern here is maybe we need to update reuirements txt file for the faker? Thanks in advance.

File: core/management/commands/generate_data.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            user.save()
+
+        for space in spaces:
+            if random.random() < 0.5:  # Adjust this probability to control how many spaces each user is assigned
+                space.members.add(user)
+                space.moderators.add(user)
+                space.save()
+
+    # Create posts with your specified data
+    posts_data = [
```
Comment (ceydaduzgec) [inline core/management/commands/generate_data.py]: After this line the code is not indented as previously. Could you add on more tab to the code below please?

**Comment by ceydaduzgec** [comment]
> Hello Ceyda I fixed the code and test it working fine also fixed read me file. One concern here is maybe we need to update reuirements txt file for the faker? Thanks in advance.

Hey, great work thanks!  We don't need to add faker since we were already using it other files. Only thing is this command is not working properly.  Could you find out the issue here please? If you want I can help you with that.

image

**Comment by sozkan4** [comment]
@ceydaduzgec I think the problem is about model filtering and I think I fix it. Could you please give it a try?

**Comment by ceydaduzgec** [comment]
@sozkan4 Looks amazing thanks a lot!

**Review by ceydaduzgec — APPROVED** [review]

**Pull request merged**